ccm_web/server/src/canvas/canvas.controller.ts

```diff
@@ -47,15 +47,24 @@ export class CanvasController {
     @Query() query: OAuthGoodResponseQuery | OAuthErrorResponseQuery,
       @Req() req: Request, @Res() res: Response, @UserDec() user: User
   ): Promise<void> {
-    logger.debug('Comparing session to state parameter, and creating new Canvas token if matching')
-    logger.debug(`Session ID: ${req.sessionID}`)
-    logger.debug(JSON.stringify(req.session, null, 2))
-
+    logger.debug(`Query: ${JSON.stringify(query, null, 2)}`)
     if (isOAuthErrorResponseQuery(query)) {
-      logger.error(`Canvas OAuth failed  due to ${query.error}. ${query.error_description}`)
-      throw new InternalServerErrorException(query.error_description)
+      if (query.error === 'access_denied') {
+        logger.debug('User rejected Canvas OAuth; sending them back to application root...')
+        res.redirect('/')
+        return
+      }
+
+      logger.error(`Canvas OAuth failed due to ${query.error}: ${String(query.error_description)}`)
+      const message = query.error_description !== undefined
+        ? query.error_description
+        : 'Canvas OAuth error was sent without a description.'
+      throw new InternalServerErrorException(message)
```

Comment:
Does this code ever will ever hit? what are other scenarios to apart from access_denied to reach L58

Reply:
I think you added this before in PR #130 to handle a scope error. So yes? I think even still, it's good to have a general fallback for scenarios you haven't anticipated.

Reply:
Yeah.. I remember that. Agreed on fallback. 

But when I coded that it was sending the error message like ` Canvas OAuth failed due to invalid_scope: A requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner. The following scopes were requested, but not granted: url:POST|/api/v1/accounts/:account_id/account_notifications`
Not anymore. The UI just be as below
![Screen Shot 2021-09-24 at 1 55 03 PM](https://user-images.githubusercontent.com/8579775/134719608-9300ae34-9e0a-4c59-8e34-a12f60b5ff92.png)

This is a separate issue.

Reply:
I don't remember that, but I'd have to look back to see.